src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using Iot.Device.Ssd1351.Command;
+
+namespace Iot.Device.Ssd1351
+{
+    /// <summary>
+    /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+    /// light emitting diode dot-matrix graphic display system. 
+    /// </summary>
+    public partial class Ssd1351 : IDisposable
+    {
+
+        private readonly SpiDevice _spiDevice;
+        private readonly GpioController _gpioDevice;
+        private readonly int _dcPinId;
+        private readonly int _resetPinId;
+        private ColorDepth _colorDepth;
+        private ColorSequence _colorSequence;
+        private int _spiBufferSize;
+
+        private const byte SCREEN_WIDTH_PX = 128;
+        private const byte SCREEN_HEIGHT_PX = 128;
+        private const int DefaultSPIBufferSize = 0x1000;
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">>The SPI device used for communication.</param>
+        /// <param name="gpioDevice">>The GPIO controller device used for communication.</param>
+        /// <param name="dcPinId">>The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPinId">>The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        public Ssd1351(SpiDevice spiDevice, GpioController gpioDevice, int dcPinId, int resetPinId)
+        {
+            _spiDevice = spiDevice;
+            _gpioDevice = gpioDevice;
+            _dcPinId = dcPinId;
+            _resetPinId = resetPinId;
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            SPIBufferSize = DefaultSPIBufferSize;
+
+            SendCommand(new SetSegmentReMapColorDepth());
+
+        }
+
+        /// <summary>
+        /// The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions
+        /// </summary>
+        public int SPIBufferSize
```

Comment:
Do we intentionally want to allow modifications on the buffer size? or is the most common case to use the same buffer size throught the lifespan of the object. I ask because if that's the case, I would just pass this in to the constructor instead and don't allow changing it.

Reply:
O.K. I will do that.... I don't see any reason for it being a property either and so I will remove the SPIBufferSize property.

Reply:
I have added a parameter to the constructor to cover the buffer size.